docs/going-into-production.rst

```diff
@@ -0,0 +1,357 @@
+document outlines the basics you need to consider when going into production.
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+
+
+.. _prod-bootstrapping:
+
+Configure bootstrapping
+=======================
+
+The process of forming a cluster is known as *bootstrapping*. Consult the
+how-to guide on :ref:`CrateDB multi-node setups <multi_node_setup>` for an
+overview of the two different ways to bootstrap a cluster.
+
+If you have been using CrateDB for development on your local machine, there is
+a good chance you have been using :ref:`single host auto-bootstrapping
+<auto-bootstrapping>`.
+
+For improved performance and resiliency, you should run production CrateDB
+clusters with one node per host machine. To do this, you must manually
+configure the bootstrapping process by telling nodes how to:
+
+  a. :ref:`Discover other nodes <discovery>`, and
+  b. :ref:`Elect a master node <master-node-election>`
+
+This process is known as *manual bootstrapping*. See the :ref:`how-to guide
+<manual-bootstrapping>` for more information about how to bootstrap a cluster
+manually.
+
+Switching to a manual bootstrapping configuration is the first step towards
+going into production.
+
+
+.. _prod-naming:
+
+Naming
+======
+
+
+.. _prod-cluster-name:
+
+Configure a logical cluster name
+--------------------------------
+
+The `cluster.name`_ setting allows you to override the default cluster name of
+``crate``. You should use this setting to give a logical name to your cluster.
+
+For example, add this to your `configuration`_ file:
+
+.. code-block:: yaml
+
+    cluster.name: acme-prod
+
+The ``acme-prod`` name suggests that this cluster is the production cluster for
+the *Acme* organization. If *Acme* has a cluster running in a staging
+environment, you might want to name it ``acme-staging``. This way, you can
+differentiate your clusters by name (visible in the `Admin UI`_).
+
+.. TIP::
+
+    A node will refuse to join a cluster if the respective cluster names
+    do not match
+
+.. SEEALSO::
+
+    :ref:`Cluster names for multi-node setups <multi-node-cluster-name>`
+
+
+.. _prod-node-names:
+
+Configure logical node names
+----------------------------
+
+By default, CrateDB sets the node name for you. You can override this by
+configuring an explicit node name using the `node.name`_ setting in your
+`configuration`_ file.
+
+CrateDB chooses automatic node names from the `sys.summits`_ table. This
+functionality is useful in development when the specifics of cluster
+architecture are less critical. However, when going into production, you
+should give each of your nodes a logical name.
+
+CrateDB supports `multiple types of node`_, determined by the ``node.master``
+and ``node.data`` settings. You can use this to name your nodes according to
+their type. For instance:
+
+.. SEEALSO::
+
+    :ref:`Node names for multi-node setups <multi-node-node-name>`
+
+
+.. _prod-node-md:
+
+Multi-purpose nodes
```

Comment:
Why is this a sub section of `Configuring logical node names`?
Doesn't make sense to me.

Reply:
because when configuring local node names per the system I have suggested, there are four types of nodes. I had to come up with descriptive names for each of the four types of node. the first node type I cover is the one with `master` and `data` set to `true`. that makes it a multi-purpose node

can you suggest a better description for this type of node? "master-eligible" doesn't cover it because you can have master-eligible nodes that don't have `data` set to true

to understand this section header you need to see it in context of the other three headers:

- Multi-purpose nodes
- Request handling and query execution nodes
- Cluster management nodes
- Request handling nodes

Reply:
ok, somehow read it wrong.